roles/branch-validation/tasks/main.yml

```diff
@@ -155,7 +167,7 @@
     that: ping.user_input|bool
 
 - include: ha.yml
-  when: validate_ha_environment
+  when: not groups['compute'] | intersect(groups['controller']) | length | bool
```

Comment:
the point of this is for one thing:  when an environment has converged controllers, compute nodes that are also controllers, then we'll run the HA tests.

This is possible because instances are launched (one each for each compute note) and scheduled on each respective compute node.  This validates scheduling and ensures that if an environment has converged controllers, that we're not missing them.  

So, if the intersection of `controller` and `compute` has a non-zero length, we have converged controllers, and we should test them by running ha.yml.